va/va.h

```diff
@@ -1126,7 +1129,7 @@ VAStatus vaDestroySurfaces (
  *  picture_height: coded picture height
  *  flag: any combination of the following:
  *    VA_PROGRESSIVE (only progressive frame pictures in the sequence when set)
- *  render_targets: render targets (surfaces) tied to the context
+ *  render_targets: a hint for render targets (surfaces) tied to the context
```

Comment:
Making the render targets optional is a driver API change.  Is that definitely ok everywhere?

Reply:
It would be up to the driver to make them tied to the context.  intel-vaapi-driver doesn't force that.

Reply:
A driver which relies on all render targets being specified in advance will be broken if they are made optional and the user then does not supply them.  This is not just about tying the surfaces to the context, they might also be required for the context creation itself - perhaps because the targets must be known directly, or because some properties of the surfaces might be needed.

Reply:
Mesa does some slightly subtle things with it here: <https://cgit.freedesktop.org/mesa/mesa/tree/src/gallium/state_trackers/va/context.c>.  I'm not sure that the VPP test is doing anything useful, but it looks like that max_references test for H.265 might.  The surfaces themselves aren't used.

Reply:
* Even in mesa, render_targets is not used directly in vaCreateContext(). 
* Mesa developers also think max_references cannot be based on num_render_targets
> commit b4fb6d76161d86c67697cf28a221d7913b8d084d
> Author: Julien Isorce <j.isorce@samsung.com>
> Date:   Thu Nov 26 08:29:54 2015 +0000
>
>    st/va: delay decoder creation until max_references is known
>
>    In general max_references cannot be based on num_render_targets.
* It is weird for me VPP and other profiles have different requirement on render_targets in Mesa
* The old API doesn't allow two contexts use the same surface at the same time, so render_targets is required when creating a context. 
> Once a surface is bound to a given context, it can not be used to create another context

Reply:
It would be better to ask mesa developer to remove the max_references test for H.265.